app/import/robot-import.js

```diff
@@ -0,0 +1,118 @@
+const sequelize = require('../config/db')
+const { v4: uuidv4 } = require('uuid')
+const { parseDateAsDDMMYYYY } = require('../lib/date-functions')
+const { getAllDogIds } = require('../dog/get-dog')
+const { isValidRobotSchema } = require('./robot-schema')
+const { addPerson } = require('../../app/person/add-person')
+const { getPersonType, getBreed } = require('../../app/lookups')
+const { dbCreate } = require('../../app/lib/db-functions')
+
+let xlBullyBreedId
+let existingDogIds
+let stats
+const newStatusId = 1
+
+const processRobotImport = async (dogsAndPeople) => {
+  stats = {
+    errors: [],
+    created: []
+  }
+
+  try {
+    /// Validate schema
+    const validationErrors = isValidRobotSchema(dogsAndPeople)
+    if (validationErrors.error !== undefined) {
+      validationErrors.error.details.map(x => stats.errors.push(x.message))
+      return { stats }
+    }
+
+    xlBullyBreedId = (await getBreed('XL Bully')).id
+
+    // Construct set of which dog ids already exist
+    existingDogIds = new Set((await getAllDogIds()).map((x) => `${x.id}`))
+
+    await sequelize.transaction(async (t) => {
+      const createdDogIds = []
+      for (const row of dogsAndPeople.data) {
+        for (const dog of row.dogs) {
+          if (stats.errors.length === 0) {
+            createdDogIds.push(await processDog(dog, t))
+          }
+        }
+
+        for (const person of row.people) {
+          if (stats.errors.length === 0) {
+            await processPerson(person, createdDogIds, t)
+          }
+        }
+
+        if (stats.errors.length > 0) {
+          throw new Error('Processing aborted')
+        }
+      }
+    })
+  } catch (e) {
+    console.log(e)
+    stats.errors.push(`Error: ${e.message}`)
+  }
+  return { stats }
+}
+
+const mapDogFields = (dog) => ({
+  id: parseInt(dog.indexNumber),
+  orig_index_number: dog.indexNumber,
+  name: dog.name,
+  birth_date: parseDateAsDDMMYYYY(dog.dateOfBirth),
+  colour: dog.colour,
+  sex: dog.sex,
+  microchip_number: dog.microchipNumber,
+  dog_breed_id: xlBullyBreedId,
+  status_id: newStatusId,
+  exported: false,
+  dog_reference: uuidv4()
+})
+
+const mapPersonFields = (person) => ({
+  id: person.id,
```

Comment:
Where is person.id being generated has this already been generated by Postgres?

Reply:
Yes, auto-increment by Postgres